src/useImageDimensions.ts

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+/**
+ * @param source either a remote URL or a local file resource.
+ * @returns original image width and height.
+ */
+function useImageDimensions(source: ImageRequireSource | URISource) {
+  const [state, setState] = useState<{
+    width?: number
+    height?: number
+    loading?: boolean
+    error?: any
+  }>({})
+  useEffect(() => {
+    if (typeof source === 'object' && typeof source.uri === 'string') {
+      setState({loading: true})
+      Image.getSize(
+        source.uri,
+        (width, height) => setState({width, height}),
+        error => setState({error}),
+      )
+    } else if (typeof source === 'number') {
+      setState(Image.resolveAssetSource(source))
```

Comment:
Since `Image.resolveAssetSource` is a synchronous method we could provide this as the initial value instead of having the first render with no dimension data, and the directly after re-rendering with the data set

Reply:
The main purpose of that module is to get dimensions for remote images or if there is some mixed images sequence: local and remote assets. It is working for local assets just additionally.
If you are using that module, so any way you should regard the loading state in a component.
Also, Image.resolveAssetSource takes some time to execute, so I think it works better with "useEffect" feature.
If I will set the result of Image.resolveAssetSourc as initial value - it will execute 2 times. First - as the initial value and second - with "useEffect".
If you are working with local assets without any remote assets, so there is no reason to use that hook anyway.

Reply:
Thanks for your comments